packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/markdownFeatures.ts

```diff
@@ -0,0 +1,150 @@
+            // runAsync is here to allow the event to complete.
+            editor.runAsync(() => {
+                handleMarkdownEvent(event, editor, triggerCharacter, elementTag);
+            });
+        },
+    };
+}
+
+function cacheGetRangeForMarkdownOperation(
+    event: PluginKeyboardEvent,
+    editor: Editor,
+    triggerCharacter: string
+): Range {
+    return cacheGetEventData(event, 'MARKDOWN_RANGE', () => {
+        const searcher = cacheGetContentSearcher(event, editor);
+
+        let startPosition: NodePosition;
+        let endPosition: NodePosition;
+        searcher.forEachTextInlineElement(textInlineElement => {
+            if (endPosition && startPosition) {
+                return true;
+            }
+            const inlineTextContent = textInlineElement.getTextContent();
+
+            // special case for immediately preceeding character being whitespace
+            if (inlineTextContent[inlineTextContent.length - 1].trim().length == 0) {
+                return false;
+            }
+
+            // special case for consecutive trigger characters
+            if (inlineTextContent[inlineTextContent.length - 1] === triggerCharacter) {
+                return false;
+            }
+
+            endPosition = textInlineElement.getStartPosition().move(inlineTextContent.length);
+            if (inlineTextContent[0] == triggerCharacter) {
+                startPosition = textInlineElement.getStartPosition();
+            } else {
+                let contentIndex = inlineTextContent.length - 1;
+                for (; contentIndex > 0; contentIndex--) {
+                    if (startPosition) {
+                        return true;
+                    }
+                    if (
+                        inlineTextContent[contentIndex] == triggerCharacter &&
+                        inlineTextContent[contentIndex - 1].trim().length == 0
+                    ) {
+                        startPosition = textInlineElement.getStartPosition().move(contentIndex);
+                        return true;
+                    }
+                }
+            }
+        });
+        return !!startPosition && !!endPosition && createRange(startPosition, endPosition);
+    });
+}
+
+function handleMarkdownEvent(
+    event: PluginKeyboardEvent,
+    editor: Editor,
+    triggerCharacter: string,
+    elementTag: string
+) {
+    editor.performAutoComplete(() => {
+        const range = cacheGetRangeForMarkdownOperation(event, editor, triggerCharacter);
+        if (!!range) {
+            // modify the range here to include the new * character
+            range.setEnd(range.endContainer, range.endOffset + 1);
+            const elementToWrap = editor.getDocument().createElement(elementTag);
+            elementToWrap.appendChild(range.extractContents());
+            elementToWrap.innerText = elementToWrap.innerText.slice(
+                1,
+                elementToWrap.innerText.length - 1
```

Comment:
A cornor case: if I type
[Star]text1[Ctrl+I]text2[Ctrl+I]text3[Star]
That only "text1" got bolded, and cursor is stopped between text1 and text2.

Reply:
I can't even get that case to work. Debugging...

Reply:
Figured it out. Some mixup between last and first. Also was able to clean it up a bit more, as we were, as a side effect nuking any currently applied styles.